app/lib/admin/actions/package_reserve.dart

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:pub_dev/package/backend.dart';
+import 'package:pub_dev/package/models.dart';
+import 'package:pub_dev/shared/datastore.dart';
+
+import 'actions.dart';
+
+final packageReserve = AdminAction(
+  name: 'package-reserve',
+  summary: 'Creates a ReservedPackage entity.',
+  description: '''
+Reserves a package name that can be claimed by @google.com accounts or
+the allowed list of email addresses.
+
+The action can be re-run with the same package name. In such cases the provided
+email list will be added to the existing ReservedPackage entity.
+''',
+  options: {
+    'package': 'The package name to be reserved.',
+    'emails': 'The list of email addresses, separated by comma.'
```

Comment:
Should we add an option to delete the entity? Maybe it is not worth it...

Reply:
We could just add:
 * `package-reserve-list`
 * `package-reserve-delete`

Life is easier, if we split them into multiple actions. We can still put them all in the same file if want that.

We could even consider naming them: 
 `package-reservation-create`
 `package-reservation-list`
 `package-reservation-delete`

naming is a nit, but it's probably more intuitive to have separate options. And it's probably easier to read the code too.

Reply:
Like a `--delete` flag?

Reply:
Yeah - probably `--delete=true` as we didn't implement real flags for actions,

Reply:
a separate command is MUCH easier to do.

Reply:
it's really just something like:
```dart
final package = options['package'];
InvalidInputException.check(
  package != null && package.isNotEmpty,
  '`package` must be given',
);

await withRetryTransaction(dbService, (tx) async {
  final existing = await tx.lookupOrNull<ReservedPackage>(
      dbService.emptyKey.append(ReservedPackage, id: package));
      
  if (exiting == null) {
    return {'message': 'ReservedPackage for "$package" does not exist!'};
  }
  
  tx.delete(existing.key);
});

return {'message': 'ReservedPackage for "$package" deleted.'};
```

A `--delete=true` flag requires a lot more explanation in the documentation, it's not very intuitive.